docs/cli-arguments.md

```diff
@@ -42,12 +42,16 @@ Usage of ./kube-state-metrics:
       --metric-blacklist string                     Comma-separated list of metrics not to be enabled. This list comprises of exact metric names and/or regex patterns. The whitelist and blacklist are mutually exclusive.
       --metric-whitelist string                     Comma-separated list of metrics to be exposed. This list comprises of exact metric names and/or regex patterns. The whitelist and blacklist are mutually exclusive.
       --namespace string                            Comma-separated list of namespaces to be enabled. Defaults to ""
+      --pod string                                  Name of the pod. When set, it is expected that --pod and --pod-namespace are both set. This is used for auto-detecting sharding. If set, this has preference over statically configured sharding. This is experimental, it may be removed without notice.
```

Comment:
Name of the pod isn't too clear. We need to mention that this is the pod that corresponds to the ksm binary itself.

Thoughts on this - "Name of the pod that contains the kube-state-metrics container"?

Reply:
Yes I think that's a good idea. I also adapted the comment to additionally say that people are most likely going to need to do this via the downward api.